spec/support/saml_response_helper.rb

```diff
@@ -18,10 +25,27 @@ def input_id
       'SAMLResponse'
     end
 
-    def response_doc
-      return Nokogiri::XML(xml_response) if @test_type == 'feature'
+    def raw_xml_response
+      if @test_type == 'feature'
+        xml_response
+      else
+        @response.body
+      end
+    end
 
-      Nokogiri::XML(@response.body)
+    def response_doc
+      if raw_xml_response =~ /EncryptedData/
+        @original_encrypted = true
+        Nokogiri::XML(
+          OneLogin::RubySaml::Response.new(
+            raw_xml_response,
+            settings: sp1_saml_settings
+          ).decrypted_document.to_s
```

Comment:
Thanks @awong-dev 

The `decrypted_document` is an instance of `XMLSecurity::SignedDocument` which is a subclass of `REXML::Document`.